### PR TITLE
OXT-1629: Backport critical CVEs from master OE layers

### PR DESCRIPTION
**Pull request by crogers1**
This PR includes several critical CVE patches and upgrades (which include CVEs) to a number of packages built and/or deployed in OpenXT.  They are all backported and adapted from upstream commits in meta-openembedded and openembedded-core.  Each package has its own commit and is marked with the OXT-1629 ticket so the backport can be tracked and removed when we upgrade OE to Zeus or newer.

`Signed-off-by: Chris Rogers <rogersc@ainfosec.com>`

### CODE REVIEW

**Comment by dozylynx** [comment]
Thanks for including the ticket number in the commits. Please could you provide the Signed-off-by with your full name?

**Comment by crogers1** [comment]
> Thanks for including the ticket number in the commits. Please could you provide the Signed-off-by with your full name?

I assumed you meant in the PR description, but were you referring to the commits themselves?

**Comment by dozylynx** [comment]
> > Thanks for including the ticket number in the commits. Please could you provide the Signed-off-by with your full name?
> 
> I assumed you meant in the PR description, but were you referring to the commits themselves?

Yes, in the commits themselves, thanks.

**Comment by crogers1** [comment]
Updated commit signoff as requested.

File: recipes-support/sqlite/sqlite3_%.bbappend [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+LICENSE = "PD"
+LIC_FILES_CHKSUM = "file://sqlite3.h;endline=11;md5=786d3dc581eff03f4fd9e4a77ed00c66"
+
+SRC_URI = "\
+  http://www.sqlite.org/2017/sqlite-autoconf-${SQLITE_PV}.tar.gz \
+  "
+SRC_URI[md5sum] = "e262a28b73cc330e7e83520c8ce14e4d"
+SRC_URI[sha256sum] = "3814c6f629ff93968b2b37a70497cfe98b366bf587a2261a56a5f750af6ae6a0"
+
```
[the remark this answers is not in the document]
Comment (crogers1) [reply recipes-support/sqlite/sqlite3_%.bbappend]: I'd have to copy over the .inc as well, which I thought was too much code duplication.  During testing I found `require` to only search within the same layer (so I can't `require` the .inc in oe-core from xenclient-oe). With a .bbappend I can update the PV, hashes, and patches to include while also getting the `require`.

**Comment by crogers1** [comment]
@eric-ch not sure about the libgcrypt warning.  I didn't see it on my stable-9 dev build with these CVEs applied.

**Comment by eric-ch** [comment]
Build [$6635](http://openxt-builder.ainfosec.com:8010/builders/openxt64/builds/6635)

File: recipes-support/sqlite/sqlite3_%.bbappend (hunk 1 above)
[the remark this answers is not in the document]
Comment (crogers1) [reply recipes-support/sqlite/sqlite3_%.bbappend]: Damn, had the require path wrong.  Fixing...

[the remark this answers is not in the document]
Comment (eric-ch) [reply recipes-support/sqlite/sqlite3_%.bbappend]: Thanks!

**Comment by jean-edouard** [comment]
Built here: http://openxt-builder.ainfosec.com:8010/builders/openxt64/builds/6640

**Comment by eric-ch** [comment]
Builds [$6646](http://openxt-builder.ainfosec.com:8010/builders/openxt64/builds/6646)

**Comment by jean-edouard** [comment]
LGTM, merging soon.

**Pull request merged**